src/plugins/intel_gpu/src/graph/impls/onednn/primitive_onednn_base.h

```diff
@@ -499,6 +489,11 @@ struct typed_primitive_onednn_impl : public typed_primitive_impl<PType> {
                 }
                 throw;    // rethrowing dnnl::error if not out_of_memory
             }
+
+            // Enqueue marker with empty events wait list (which will trigger wait for all previously enqueued tasks) and
+            // return it as oneDNN primitive's event as it is a single option for proper synchronization
+            if (instance.is_output_event())
+                event = stream.enqueue_marker({});
```

Comment:
Is it to ensure that all onednn kernels are done before CPU layer execution, not for network output? If so, could you describe that on comment? I think it is unclear why it is necessary..

Reply:
Yes, in most cases we need it to ensure that onednn kernel is finished before CPU layer execution, but also event with marker will be created if onednn primitive will be network's output (it may be used in loop primitive for networks synchronization, and maybe in some other cases I'm not aware of...)
And I think we can try to use it for get_memory() or reset_network() methods for in_order queue - in such a case we will avoid clFinish() call at all which theoretically may improve performance a bit (if I'm not mistaken, clFinish() call flushes caches, and maybe has some other side-effects)

Reply:
@isanghao, I have updated comment and mentioned cases when this event is used

Reply:
Hmm basically I agree with removing clfinish as much as possible. 
However I have a curiosity about "flush cache" you mentioned, becuase 
previously I experienced cache inconsistency b/w CPU and dGPU on host memory even though I did clFinish.